connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -0,0 +1,300 @@
+            .setProperty(ExchangeProperty.OFFSET, constant(0))
+            .setProperty(ExchangeProperty.LIMIT, constant(this.emailConnectorConfig.getRbacElementsPerPage()))
+            // Begin fetching the users...
+            .loopDoWhile(this.notFinishedFetchingAllPages)
+                // Prepare the request and its headers.
+                .process(this.rbacPrincipalsRequestPreparerProcessor)
+                .to(this.emailConnectorConfig.getRbacURL())
+                // Process the results and determine if we should keep looping.
+                .process(this.rbacUsersProcessor)
+            .end()
+            .process(this.recipientsFilter)
+            .to(direct(Routes.SEND_EMAIL_BOP_CHOICE));
+
+        /*
+         * Fetches the users from IT and filters them.
+         */
+        from(direct(Routes.FETCH_USERS_IT))
+            .routeId(Routes.FETCH_USERS_IT)
+            // Clear all the headers that may come from the previous route.
+            .removeHeaders("*")
+            // Initialize the offset and the loop condition so that we at least
+            // enter once on it.
+            .setProperty(ExchangeProperty.OFFSET, constant(0))
+            .setProperty(ExchangeProperty.LIMIT, constant(this.emailConnectorConfig.getItElementsPerPage()))
+            // Begin fetching the users...
+            .loopDoWhile(this.notFinishedFetchingAllPages)
+                .process(this.itUserRequestPreparer)
+                .to(this.setUpITEndpoint(this.getCamelContext()))
+                .process(this.itResponseProcessor)
+            .end()
+            .process(this.recipientsFilter)
+            .to(direct(Routes.SEND_EMAIL_BOP_CHOICE));
+
+        /*
+         * Fetches an RBAC group. If it doesn't exist for some reason, we
+         * simply fetch all the users as normal. If it exists, we just fetch
+         * the users from that particular group.
+         */
+        from(direct(Routes.FETCH_GROUP))
+            .routeId(Routes.FETCH_GROUP)
+            // Clear all the headers that may come from the previous route.
+            .removeHeaders("*")
+            .process(this.rbacGroupRequestPreparerProcessor)
+            .to(this.emailConnectorConfig.getRbacURL())
+            .process(this.rbacGroupProcessor)
+            .choice()
+                .when(simpleF("${exchangeProperty.%s}", RBACConstants.RBAC_GROUP_IS_PLATFORM_DEFAULT))
+                    .to(direct(Routes.FETCH_USERS))
+                .otherwise()
+                    .to(direct(Routes.FETCH_GROUP_USERS))
+            .end()
+            // When a 404 "Not Found" status code is received, then we simply
+            // continue with the execution.
+            .onException(HttpOperationFailedException.class)
+                .onWhen(simpleF("${exception.statusCode} == %d", HttpStatus.SC_NOT_FOUND))
+                    .handled(true)
+                    .setProperty(ExchangeProperty.USERNAMES, constant(new HashSet<String>()));
```

Comment:
You're right, this is a leftover from the old code. I've noticed, though, that the `camel-bean` dependency is required in order to do the `${exception.statusCode}` thing in this same code block... I'm writing a predicate to see if we can replace that to avoid having to use that dependency.